iocore/dns/DNS.cc

```diff
@@ -474,7 +474,7 @@ DNSHandler::open_con(sockaddr const *target, bool failed, int icon, bool over_tc
   ip_port_text_buffer ip_text;
   PollDescriptor *pd = get_PollDescriptor(dnsProcessor.thread);
 
-  if (!icon && target) {
+  if (!icon && target && &ip.sa != target) {
```

Comment:
Wouldn't it be better to move this test to ats_ip_copy() ?

Reply:
Yes.

Reply:
you're right, i don't like c. I shouldn't mimic it.